pdata/README.md

```diff
@@ -0,0 +1,144 @@
+declared structs. We keep a pointer to OTLP protobuf in the "orig" member field. This allows efficient translation 
+to/from OTLP wire protocol. The underlying data structure is kept private so that we are free to make changes to it 
+in the future.
+
+API provided by pdata ensures that no underlying data can be used in more than one place. Each pdata instance cannot 
+contain a reference to an object that is already used somewhere else.
+
+## API naming convention
+
+### protobuf message representation in pdata
+
+Pipeline data structs SHOULD be based on the names of the underlying OTLP protobuf messages. Data types for 
+protobuf messages defined as part of another message SHOULD include the owner's name as a prefix. Exceptions to these 
+rules are possible, but not encouraged. Another name can be chosen for brevity or if the struct provides a 
+different interface to manipulate the underlying data. The following exceptions are currently accepted:
+
+- `plog.Logs` based on `LogsData` protobuf message.
+- `pmetric.Metrics` based on `MetricsData` protobuf message.
+- `ptrace.Traces` based on `TracesData` protobuf message.
+- `pcommon.Slice` based on `ArrayValue` protobuf message.
+- `pcommon.Map` based on `KeyValueList` protobuf message.
+- `pcommon.Value` based on `AnyValue` protobuf message.
+
+Each pdata struct MUST have an initialization function starting with `New` prefix. Usage of zero-initialized is 
+prohibited and can case panic. Each pdata struct MUST provide the following methods:
+
+- `MoveTo(<type>)`: moves all the data from one struct instance to another. Destination instance is overridden, the 
+  source instance is re-initialized.
+- `CopyTo(<type>)`: copies all the data from one struct instance to another. Destination instance is overridden, the
+  source instance is not modified.
+
+### protobuf fields representation in pdata
+
+### Singular fields
+
+Each pdata struct based on a protobuf message SHOULD have getter methods for every protobuf field called the same as
+the protobuf field but in CamelCase. Exceptions to this rule are allowed if exposing a field is not desirable in pdata. 
+pdata can also provide a different interface to manipulate fields without exposing them explicitly as it's done in 
+`pcommon.Map`. If protobuf field is of scalar or enum type, the corresponding pdata struct MUST provide a setter 
+method with `Set` prefix.
+
+Some fields of scalar type in protobuf message MAY be represented by another pdata type providing an additional 
+interface, for example `pcommon.Timestamp`.
+
+### Optional fields
+
+Each optional field in protobuf message exposed in pdata MUST have an additional method with `Has` prefix to
+determine if the field is set.
+
+### OneOf fields
+
+A pdata struct representing a protobuf message with an oneof field MUST provide getter and setter methods for each 
+option of the oneof field. If an oneof field option is another protobuf message, setter name MUST include `Empty` 
```

Comment:
'oneof' starts with a consonant sound /w/ so it should be preceded by `a`, not `an` (see [here](https://english.stackexchange.com/a/333972))
```suggestion
A pdata struct representing a protobuf message with a oneof field MUST provide getter and setter methods for each 
option of the oneof field. If a oneof field option is another protobuf message, setter name MUST include `Empty` 
```